bagel/cli.py

```diff
@@ -335,3 +346,163 @@ def bids(
 
     print(f"Saved output to:  {output}")
+
+
+@bagel.command()
+def derivatives(
+    tabular: Path = typer.Option(
+        ...,
+        "--tabular",
+        "-t",
+        help="The path to a .tsv containing subject-level processing pipeline status info. Expected to comply with the Nipoppy processing status file schema.",
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        resolve_path=True,
+    ),
+    # TODO: Remove _path?
+    jsonld_path: Path = typer.Option(
+        ...,
+        "--jsonld-path",
+        "-p",  # for pheno
+        help="The path to a .jsonld file containing the phenotypic data for your dataset, created by the bagel pheno command. This JSONLD may optionally also include the BIDS metadata for the dataset (created by the bagel bids command).",
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        resolve_path=True,
+    ),
+    output: Path = typer.Option(
+        "pheno_derivatives.jsonld",
+        "--output",
+        "-o",
+        help="The path for the output .jsonld file.",
+        file_okay=True,
+        dir_okay=False,
+        resolve_path=True,
+    ),
+    overwrite: bool = typer.Option(
+        False,
+        "--overwrite",
+        "-f",
+        help="Overwrite output file if it already exists.",
+    ),
+):
+    """
+    Extract subject processing pipeline and derivative metadata from a tabular processing status file and
+    integrate them in a single .jsonld with subjects' harmonized phenotypic data (from the bagel pheno command) and optionally,
+    BIDS metadata (from the bagel bids command).
+    NOTE: Must be run AFTER the pheno command.
+
+    This command will create a valid, subject-level instance of the Neurobagel
+    graph data model for the combined metadata in the .jsonld format.
+    You can upload this .jsonld file to the Neurobagel graph.
+    """
+    futil.check_overwrite(output, overwrite)
+
+    space = 51
+    print(
+        "Running initial checks of inputs...\n"
+        f"   {'Existing subject graph data to augment (.jsonld):' : <{space}}{jsonld_path}\n"
+        f"   {'Processing status file (.tsv):' : <{space}}{tabular}"
+    )
+
+    # Load and do basic TSV validation of the processing status file
+    status_df = futil.load_tabular(tabular, input_type="processing status")
+
+    # Check for missing participant IDs
+    if row_indices := putil.get_rows_with_empty_strings(
+        status_df, [PROC_STATUS_COLS["participant"]]
+    ):
+        raise LookupError(
+            f"Your processing status file contains missing values in the column '{PROC_STATUS_COLS['participant']}'. "
+            "Please ensure that every row has a non-empty participant id. "
+            f"We found missing values in the following rows (first row is zero): {row_indices}."
+        )
+
+    pipelines = status_df[PROC_STATUS_COLS["pipeline_name"]].unique()
+    dutil.check_pipelines_are_recognized(pipelines)
```

Comment:
I think I prefer the current approach because the logic is more explicit, the individual function has fewer things to worry about + a simpler input, and thus is also easier to test. At the moment, we need to extract the unique pipeline names anyways to check that their corresponding versions in the TSV are also recognized, via `check_pipeline_versions_are_recognized()`. We could combine all these steps into a single validation function, but again, I think this might sacrifice test maintainability/readability as well as modularity.

If you have a strong preference for refactoring out this validation entirely, maybe we can open a separate issue to discuss it?

Reply:
agree that this would be in a separate issue - and no strong opinion on whether that's needed. Just to be clear: I'd keep the `check_pipelines_are_recognized` around. I'd just have a new `return_recognized_pipelines` function that we feed the whole `status_df` to and that internally handles looking up the right column name, getting the unique values from it, running them into check_pipelines to weed out the evil doers / or just error out, and then returns the rest. 

So essentially the same syntax you have here, same modularity, just a bit less stuff happening directly in the CLI. Should be pretty testable too